src/hotspot/share/interpreter/abstractInterpreter.cpp

```diff
@@ -110,36 +110,61 @@ AbstractInterpreter::MethodKind AbstractInterpreter::method_kind(const methodHan
-    vmIntrinsics::ID id = m->intrinsic_id();
-    assert(MethodHandles::is_signature_polymorphic(id), "must match an intrinsic");
-    MethodKind kind = (MethodKind)(method_handle_invoke_FIRST +
-                                   vmIntrinsics::as_int(id) -
-                                   static_cast<int>(vmIntrinsics::FIRST_MH_SIG_POLY));
-    assert(kind <= method_handle_invoke_LAST, "parallel enum ranges");
-    return kind;
-  }
+  vmIntrinsics::ID id = m->intrinsic_id();
+  if (id != vmIntrinsics::_none) {
+    if (m->is_method_handle_intrinsic()) {
+      assert(MethodHandles::is_signature_polymorphic(id), "must match an intrinsic");
+      MethodKind kind = (MethodKind)(method_handle_invoke_FIRST +
+                                    vmIntrinsics::as_int(id) -
+                                    static_cast<int>(vmIntrinsics::FIRST_MH_SIG_POLY));
+      assert(kind <= method_handle_invoke_LAST, "parallel enum ranges");
+      return kind;
+    }
 
 #ifndef ZERO
-  switch (m->intrinsic_id()) {
-    // Use optimized stub code for CRC32 native methods.
-    case vmIntrinsics::_updateCRC32            : return java_util_zip_CRC32_update;
-    case vmIntrinsics::_updateBytesCRC32       : return java_util_zip_CRC32_updateBytes;
-    case vmIntrinsics::_updateByteBufferCRC32  : return java_util_zip_CRC32_updateByteBuffer;
-    // Use optimized stub code for CRC32C methods.
-    case vmIntrinsics::_updateBytesCRC32C             : return java_util_zip_CRC32C_updateBytes;
-    case vmIntrinsics::_updateDirectByteBufferCRC32C  : return java_util_zip_CRC32C_updateDirectByteBuffer;
-    case vmIntrinsics::_intBitsToFloat:      return java_lang_Float_intBitsToFloat;
-    case vmIntrinsics::_floatToRawIntBits:   return java_lang_Float_floatToRawIntBits;
-    case vmIntrinsics::_longBitsToDouble:    return java_lang_Double_longBitsToDouble;
-    case vmIntrinsics::_doubleToRawLongBits: return java_lang_Double_doubleToRawLongBits;
-    default:                                 break;
-  }
+    switch (m->intrinsic_id()) {
+      // Use optimized stub code for CRC32 native methods.
+      case vmIntrinsics::_updateCRC32: return java_util_zip_CRC32_update;
+      case vmIntrinsics::_updateBytesCRC32: return java_util_zip_CRC32_updateBytes;
+      case vmIntrinsics::_updateByteBufferCRC32: return java_util_zip_CRC32_updateByteBuffer;
+      // Use optimized stub code for CRC32C methods.
+      case vmIntrinsics::_updateBytesCRC32C: return java_util_zip_CRC32C_updateBytes;
+      case vmIntrinsics::_updateDirectByteBufferCRC32C: return java_util_zip_CRC32C_updateDirectByteBuffer;
+      case vmIntrinsics::_intBitsToFloat: return java_lang_Float_intBitsToFloat;
+      case vmIntrinsics::_floatToRawIntBits: return java_lang_Float_floatToRawIntBits;
+      case vmIntrinsics::_longBitsToDouble: return java_lang_Double_longBitsToDouble;
+      case vmIntrinsics::_doubleToRawLongBits: return java_lang_Double_doubleToRawLongBits;
```

Comment:
Could you align these return statements?

Reply:
Ok, I'll try tidying up - is there some rule of thumb on how to align in a switch like this where the longest case label is.. pretty long?